ignite/pkg/cosmosgen/generate_typescript.go

```diff
@@ -55,8 +57,12 @@ func (g *generator) generateTS() error {
 	// modules when the root templates are re-generated.
 	for _, modules := range g.thirdModules {
 		data.Modules = append(data.Modules, modules...)
+		for _, m := range modules {
+			if strings.HasPrefix(m.Pkg.Name, "cosmos.staking") {
+				data.IsConsumerChain = false
+			}
```

Comment:
What do you think about leaving the `IsConsumerChain` with the default `false` value instead of initialising it to `true` only when creating the payload and setting it to true when `interchain_security.ccv.consumer` module is available?

```suggestion
			if strings.HasPrefix(m.Pkg.Name, "interchain_security.ccv.consumer") {
				data.IsConsumerChain = true
			}
```

I'm not sure yet but isn't it possible to have the `staking` module enabled in a consumer chain when using the `ccv` one?

Reply:
default `IsConsumerChain` = false really makes sense 👍 will try this one. As for staking module in consumer chain - No, consumer chain doesn't have staking at all (as it doesn't have own validators)